### ISSUE TITLE
mypy.ini processing should allow variable expansion ("interpolation")

### ISSUE THREAD

**Issue by jtmoon79** [python/mypy#6476]
**tl;dr** I cannot write variable-expanded directory paths within `mypy.ini` to set the `mypy_path`.  I cannot commit a useful `mypy.ini`.  It would be cool if _mypy_ supported variable expansion ("interpolation") within it's `configparser` processed file.

<br/>

### Problem example

Given my pipenv-created virtualenv at `C:/Users/user1/.virtualenvs/project1-rAPKj25Z`, I want to set `mypy_path` to the virtual environment's `Lib/site-packages`.  I can set

    [mypy]
    mypy_path: C:/Users/user1/.virtualenvs/project1-rAPKj25Z/Lib/site-packages

but that is not portable (i.e. committable).  I would like to write something portable like

    [mypy]
    mypy_path: %(VIRTUAL_ENV)s/Lib/site-packages

where `VIRTUAL_ENV` is set by `pipenv` within the caller's environment (or perhaps a `python-dotenv` file).

### ...

This is a _feature request_.  I am not sure if this is outside the expectations for `mypy.ini` files.  However, it seems like a clean way to have more portable `mypy.ini` files.

### Using

- Windows 10 Pro
- Python 3.7.1
- mypy 0.670
- pipenv 2018.11.26

**Comment by jtmoon79** [comment]
I thought this could easily be solved by

    --- a/mypy/main.py                                                                                                                                                                                                                                                                      
    +++ b/mypy/main.py
    @@ -905,7 +905,7 @@ def parse_config_file(options: Options, filename: Optional[str]) -> None:                                                                                                                                                                                        
    else:
    config_files = tuple(map(os.path.expanduser, defaults.CONFIG_FILES))                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
    -    parser = configparser.RawConfigParser()                                                                                                                                                                                                                                        
    +    parser = configparser.ConfigParser(os.environ)

_Glancing_ through [nearby code](https://github.com/python/mypy/blob/86515b5e129aad55d990c08cfb7d8aeb8f847811/mypy/main.py#L908), there doesn't appear to be a problem with such a change.  I tried out the change and it resulted in numerous errors `mypy.ini: [mypy]: Unrecognized option:`, e.g.

    mypy.ini: [mypy]: Unrecognized option: venv_path = %(VIRTUAL_ENV)s
    mypy.ini: [mypy]: Unrecognized option: aclocal_path = C:\Programs\Git\mingw64\share\aclocal;C:\Programs\Git\usr\share\aclocal
    mypy.ini: [mypy]: Unrecognized option: allusersprofile = C:\ProgramData
    ...

For `mypy.ini`

    [mypy]
    venv_path: %(VIRTUAL_ENV)s
    mypy_path: %(venv_path)s/Lib/site-packages

Darn.

**Comment by JelleZijlstra** [comment]
For context, do other development tools with config files (e.g., linters) usually support interpolating environment variables? That could help decide whether this is something worth supporting in mypy.

**Comment by jtmoon79** [comment]
> do other development tools with config files (e.g., linters) usually support interpolating environment variables?

@JelleZijlstra this could be broken out into two questions:
- should mypy support interpolating the `.ini` file?
- how make `.ini` file parameter `mypy_path` more portable (i.e. committable)?

**Comment by wakemaster39** [comment]
This would be a really helpful feature for situations like pipenv where I need to have a wrapper script to make sure mypy knows where to look for packages to avoid import errors and to ensure that the config is portable for the rest of my team.

**Comment by ilevkivskyi** [comment]
This is now supported (in most cases), see https://mypy.readthedocs.io/en/latest/config_file.html:
> Some flags support user home directory and environment variable expansion. To refer to the user home directory, use `~` at the beginning of the path. To expand environment variables use `$VARNAME` or `${VARNAME}`.

**Comment by Mattwmaster58** [comment]
@ilevkivskyi 
> Some flags support user home directory and environment variable expansion

Any hint as to what flags those are?
